Khiem_Nguyen/Python_3/Python/python_stack/flask_fundamentals/dojo_survey_with_validation/server.py

```diff
@@ -0,0 +1,49 @@
+from flask import Flask, render_template, request, redirect, url_for, flash, session
+
+app = Flask(__name__)
+app.secret_key = "ThisIsSecret"
+
+@app.route("/")
+def index():
+    return render_template("index.html")
+
+@app.route("/process", methods=["POST"])
+def createUserResult():
+    if len(request.form["name"]) < 1:
+        flash("Name cannot be empty!")
+
+    if len(request.form["comment"]) < 1:
+        flash("Comments cannot be empty")
+
+    if len(request.form["comment"]) > 120:
+        flash("A comment cannot have more than 120 characters")
+
+    if "_flashes" in session.keys():
+        return redirect("/")
+
+    else:
+        session["name"] = request.form["name"]
+        session["location"] = request.form["location"]
+        session["language"] = request.form["language"]
+        session["comment"] = request.form["comment"]
+    return render_template("result.html")
```

Comment:
return redirect("/result") //Don't render since it is a POST route. In this case, the url link will be 'localhost/process' once you submit. Refreshing the link will reprocess createUserResult function